compiler/rustc_parse/src/parser/item.rs

```diff
@@ -872,16 +872,67 @@ impl<'a> Parser<'a> {
         }
     }
 
-    /// Is this an `(unsafe auto? | auto) trait` item?
-    fn check_auto_or_unsafe_trait_item(&mut self) -> bool {
-        // auto trait
-        self.check_keyword(exp!(Auto)) && self.is_keyword_ahead(1, &[kw::Trait])
-            // unsafe auto trait
-            || self.check_keyword(exp!(Unsafe)) && self.is_keyword_ahead(1, &[kw::Trait, kw::Auto])
+    /// Is this an `[ impl(in path) ]? unsafe? auto? trait` item?
```

Comment:
As said earlier the RFC put the impl restrictions after the unsafe, should also be changed.